### PR TITLE
refactor(windows): handle tray menu events in the main loop

### LINKED ISSUE #2983: windows: handle system tray menu events in Controller directly
**Is your feature request related to a problem? Please describe.**
No it's just a suggestion conectado made, which will make the code more consistent

**Describe the solution you'd like**
This function is called from the system tray handle but most of it what it does is send stuff to Controller. We should be calling it from Controller so that if there's anything in the state machine that interferes with other stuff, it'll be easy to handle. https://github.com/firezone/firezone/blob/0c25ad57cb202b7c0fd57117d0e9d6f7dd47e6e3/rust/windows-client/src-tauri/src/client/gui.rs#L166

**Describe alternatives you've considered**
If it doesn't hurt anything, we can leave it where it is for a while.

**Additional context**
None

### PR DESCRIPTION
**Pull request by ReactorScram**
Closes #2983

### CODE REVIEW

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 1]
```diff
@@ -227,53 +227,17 @@ fn handle_system_tray_event(app: &tauri::AppHandle, event: TrayMenuEvent) -> Res
         .ok_or_else(|| anyhow!("can't get Managed struct from Tauri"))?
```
Comment (thomaseizinger) [inline rust/windows-client/src-tauri/src/client/gui.rs]: Did you know that you can use `.context` also on `Option`s to turn them into `Result`s? Would be a bit simpler in this case :)

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 2]
```diff
@@ -227,53 +227,17 @@ fn handle_system_tray_event(app: &tauri::AppHandle, event: TrayMenuEvent) -> Res
...
-                win.hide()?;
-            } else {
-                win.show()?;
-            }
-        }
-        TrayMenuEvent::SignIn => ctlr_tx.blocking_send(ControllerRequest::SignIn)?,
-        TrayMenuEvent::SignOut => ctlr_tx.blocking_send(ControllerRequest::SignOut)?,
-        TrayMenuEvent::Quit => ctlr_tx.blocking_send(ControllerRequest::Quit)?,
-    }
+    ctlr_tx.blocking_send(ControllerRequest::SystemTrayMenu(event))?;
```
Comment (thomaseizinger) [inline rust/windows-client/src-tauri/src/client/gui.rs]: This variable could be inlined, not sure it adds much value if it is only used once :)

File: rust/windows-client/src-tauri/src/client/gui.rs [hunk 3]
```diff
@@ -579,11 +551,19 @@ async fn run_controller(
                     Req::GetAdvancedSettings(tx) => {
                         tx.send(controller.advanced_settings.clone()).ok();
                     }
-                    Req::Quit => break,
                     Req::SchemeRequest(url) => if let Err(e) = controller.handle_deep_link(&url).await {
                         tracing::error!("couldn't handle deep link: {e:#?}");
                     }
-                    Req::SignIn => {
+                    Req::SystemTrayMenu(TrayMenuEvent::About) => controller.toggle_window("about")?,
```
Comment (thomaseizinger) [inline rust/windows-client/src-tauri/src/client/gui.rs]: Is there no type-safe way of handling this? Should we at least extract `about` into a constant?

**Review by thomaseizinger — APPROVED** [review]
LGTM

File: rust/windows-client/src-tauri/src/client/gui.rs (hunk 3 above)
Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client/gui.rs]: That's a good idea, I'll try it

File: rust/windows-client/src-tauri/src/client/gui.rs (hunk 1 above)
Comment (ReactorScram) [reply rust/windows-client/src-tauri/src/client/gui.rs]: I think I didn't know that 2 months ago, thanks :)
Edit: I think I forgot it yesterday as well

**Pull request merged**